src/modal.ts

```diff
@@ -0,0 +1,462 @@
+            }
+            this.style.zIndex = DEFAULT_MODAL_ZINDEX;
+            if(this.stack)
+                this.style.zIndex = DEFAULT_MODAL_ZINDEX + active_modals.length;
+            else
+                active_modals.forEach((m:any) => m.hide());
+            if(!this.actualParent)
+                this.actualParent = this.parentElement;
+            this._open = this.open = true;
+            active_modals.push(this);
+            document.body.appendChild(this);
+        }
+    }
+    hide(delay?:number) {
+        this.clearTimer();
+        if (!this._open)
+            return;
+        if (delay)
+            this.timer_obj = setTimeout( () => {this.timer_obj=null;
+                                        this.hide()}, delay);
+        else
+            this.open = false;
+        if (this.animationDur == 0)
+            this.closeEnd();
+
+    }
+    closeEnd() {
+        this._open = this.open;
+        if (!this._open) {
+            this.actualParent?.appendChild(this);
+            let ix = active_modals.indexOf(this);
+            if (ix >= 0)
+                active_modals.splice(ix, 1);
+        }
+    }
+    private _dragStart(event:any) {
+        if (event.target.matches(Object.keys(this._guide_value)))
+            event.preventDefault();
+    }
+    private _touchStart(event:any) {
+        let css_pos = window.getComputedStyle(this._wrapper_el)
+                                 .getPropertyValue("position");
+        if(css_pos == "relative")
+            return;
+        if (!this._wrp_init_bounds)
+            this._wrp_init_bounds = this._wrapper_el.getBoundingClientRect();
+        for (let guide_sel of Object.keys(this._guide_value)) {
+            let el = event.target;
+            while (el) {
+                if (el.matches(guide_sel)) {
+                    this._mouse_prev_y = event.clientY;
+                    this._mouse_prev_x = event.clientX;
+                    this._mouse_move_listener = this._mouseMoveEvent.bind(this);
+                    this._mouse_up_listener = this._mouseUpEvent.bind(this);
+                    this._mouse_pos_mod = this._guide_value[guide_sel];
+                    document.addEventListener("mouseup", 
+                                             this._mouse_up_listener);
+                    document.addEventListener("touchend", 
+                                             this._mouse_up_listener);
+                    document.addEventListener("mousemove", 
+                                             this._mouse_move_listener);
+                    document.addEventListener("touchmove", 
+                                             this._mouse_move_listener);
+                    break;
+                }
+                el = el.parentElement;
+            }
+            if (this._mouse_move_listener)
+                break;
+        }
+    }
+    private _mouseMoveEvent(event:any) {
+        let xdiff = event.clientX - this._mouse_prev_x;
```

Comment:
MouseMove event is called on touchMove event where the clientX and clientY has to be read differently. You can refer the slider's touchMoveEvent to see how its set.